File: C++/2.2.Arrays_OddOccurencesInArray.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> &A){
    sort(A.begin(),A.end());
    int i = 0;
    int j = 0;
    int count = 0;
    while(count % 2 == 0){
        count = 0;
        i=j;
        while(A[j]==A[i]){
           j++;
           count++;
        }
    }
   return A[j-1]; 
}

int main(){

    vector<int> v={6};
    cout << solution(v);

return 0;
}

/* Detected time complexity: O(N) or O(N*log(N)) */
// 100 %

File: Python/2.2.Arrays_OddOccurencesInArray.py
# This one is O(N) or O(N*log(N))
# Score 100

def solution(A):
    count={}
    for i in A:
        if not i in count:
            count[i]=1
        else:
            count[i]+=1
    for i in count:
        if count[i] % 2 == 1:
            return i


A=[1,1,2,2,3,3,2,2,7,8,7,8,2]
print(solution(A))


